File: lib/core/routing/route_paths.dart
import '../constants/role.dart';

/// go_router 경로 상수
abstract final class RoutePaths {
  static const login = '/login';
  static const changePassword = '/change-password';

  // Shell 하위 경로
  static const dashboard = '/';
  static const resume = '/resume';
  static const studyRoom = '/study-room';
  static const studyRoomNotes = '/study-room/notes';
  static String studyRoomNoteSource(String sourceId, {String? noteId}) {
    final path = '/study-room/notes/$sourceId';
    if (noteId == null || noteId.isEmpty) return path;
    return Uri(path: path, queryParameters: {'noteId': noteId}).toString();
  }

  static const board = '/board';
  static const records = '/records';
  static const recordsCreate = '/records/create';
  static const recordsCreateCert = '/records/create/certification';
  static const recordsCreateStudy = '/records/create/study';
  static const recordsCreateBlog = '/records/create/blog';
  static const recordsCreateStudyCert = '/records/create/study-cert';
  static const recordsCreatePrecourseQuiz = '/records/create/precourse-quiz';
  static const mileage = '/mileage';
  static const mileageShop = '/mileage/shop';
  static const mileageCart = '/mileage/shop/cart';
  static const assessments = '/assessments';
  static const myPage = '/my-page';
  static const settings = '/settings';
  static const forms = '/forms';
  static const qualExams = '/qual-exams';
  static const resumeEdit = '/resume/:resumeId/edit';

  // Admin Shell
  static const admin = '/admin';
  static const adminRecords = '/admin/records';
  static const adminResumes = '/admin/resumes';
  static const adminBoard = '/admin/board';
  static const adminBoardNoticeCreate = '/admin/board/create';
  static const adminBoardScheduledCreate = '/admin/board/scheduled/create';
  static const adminBoardAlertPopupCreate = '/admin/board/alert-popups/create';
  static const adminStudyRoom = '/admin/study-room';
  static const adminMyPage = '/admin/my-page';
  static const adminSettings = '/admin/settings';
  static const adminStudyRoomCreate = '/admin/study-room/create';
  static const adminStudents = '/admin/students';
  static const adminStudentsCreate = '/admin/students/create';
  static const adminCohorts = '/admin/cohorts';
  static const adminCohortsCreate = '/admin/cohorts/create';
  static const adminFormTasks = '/admin/form-tasks';
  static const adminFormTasksCreate = '/admin/form-tasks/create';
  static const adminSeating = '/admin/seating';
  static const adminMileage = '/admin/mileage';
  static const adminMileageProducts = '/admin/mileage/products';
  static const adminMileageProductsCreate = '/admin/mileage/products/create';
  static const adminMileageRequests = '/admin/mileage/requests';
  static const adminMileageAdjust = '/admin/mileage/adjust';
  static const adminMileageSettings = '/admin/mileage/settings';
  static const adminInstructors = '/admin/instructors';
  static const adminInstructorsCreate = '/admin/instructors/create';
  static const adminAttendance = '/admin/attendance';
  static const adminSeatPresence = '/admin/seat-presence';
  static const adminAssessments = '/admin/assessments';
  static const adminAiQuality = '/admin/ai-quality';

  // Instructor Shell
  static const instructor = '/instructor';
  static const instructorResumes = '/instructor/resumes';
  static const instructorBoard = '/instructor/board';
  static const instructorBoardCreate = '/instructor/board/create';
  static const instructorAssessments = '/instructor/assessments';
  static const instructorAssessmentsCreate = '/instructor/assessments/create';
  static const instructorCurriculum = '/instructor/curriculum';
  static const instructorMyPage = '/instructor/my-page';
  static const instructorSettings = '/instructor/settings';

  static String assessmentTakePath(String assessmentId) =>
      '/assessments/$assessmentId/take';
  static String assessmentResultPath(String assessmentId) =>
      '/assessments/$assessmentId/result';

  static String instructorBoardNoticeEditPath(String noticeId) =>
      '/instructor/board/$noticeId/edit';
  static String instructorAssessmentDetailPath(String assessmentId) =>
      '/instructor/assessments/$assessmentId';
  static String instructorAssessmentEditPath(String assessmentId) =>
      '/instructor/assessments/$assessmentId/edit';
  static String instructorAssessmentSubmissionPath(
    String assessmentId,
    String submissionId,
  ) => '/instructor/assessments/$assessmentId/submissions/$submissionId';

  static String adminAssessmentDetailPath(String assessmentId) =>
      '/admin/assessments/$assessmentId';
  static String adminAssessmentSubmissionPath(
    String assessmentId,
    String submissionId,
  ) => '/admin/assessments/$assessmentId/submissions/$submissionId';

  static String homeFor(UserRole role) => switch (role) {
    UserRole.admin => admin,
    UserRole.instructor => instructor,
    UserRole.student => dashboard,
  };

  static String adminMileageProductEditPath(String productId) =>
      '/admin/mileage/products/$productId/edit';

  static String adminStudentDetailPath(String uid) => '/admin/students/$uid';
  static String adminStudentEditPath(String uid) => '/admin/students/$uid/edit';
  static String adminFormTaskDetailPath(String taskId) =>
      '/admin/form-tasks/$taskId';
  static String adminFormTaskEditPath(String taskId) =>
      '/admin/form-tasks/$taskId/edit';
  static String adminCohortEditPath(String cohortId) =>
      '/admin/cohorts/$cohortId/edit';

  static String resumeEditPath(
    String resumeId, {
    String? section,
    String? cohortId,
    bool openFeedback = false,
  }) {
    final path = '/resume/$resumeId/edit';
    final params = <String, String>{};
    if (section != null && section.isNotEmpty) params['section'] = section;
    if (cohortId != null && cohortId.isNotEmpty) params['cohortId'] = cohortId;
    // 목록의 「읽으러 가기」로 들어오면 종이 펼쳐진 채로 연다. 가는 곳은 같고
    // 도착 상태가 다르다.
    if (openFeedback) params['feedback'] = '1';
    if (params.isEmpty) return path;
    final query = params.entries.map((e) => '${e.key}=${e.value}').join('&');
    return '$path?$query';
  }

  static String adminBoardNoticeEditPath(String noticeId) =>
      '/admin/board/$noticeId/edit';
  static String adminBoardScheduledEditPath(String scheduledId) =>
      '/admin/board/scheduled/$scheduledId/edit';
  static String adminBoardAlertPopupEditPath(String popupId) =>
      '/admin/board/alert-popups/$popupId/edit';

  static String adminStudyRoomPackagePath(String packageId) =>
      '/admin/study-room/$packageId';
  static const attendance = '/attendance';
  static const seating = '/seating';
}

File: lms_react/src/app/routePaths.ts
import type { UserRole } from '../domain/types';

/** 경로 상수 — Flutter `core/routing/route_paths.dart` 그대로 */
export const RoutePaths = {
  login: '/login',
  changePassword: '/change-password',

  // 학생 셸
  dashboard: '/',
  resume: '/resume',
  studyRoom: '/study-room',
  studyRoomNotes: '/study-room/notes',
  board: '/board',
  records: '/records',
  recordsCreate: '/records/create',
  recordsCreateCert: '/records/create/certification',
  recordsCreateStudy: '/records/create/study',
  recordsCreateBlog: '/records/create/blog',
  recordsCreateStudyCert: '/records/create/study-cert',
  recordsCreatePrecourseQuiz: '/records/create/precourse-quiz',
  mileage: '/mileage',
  mileageShop: '/mileage/shop',
  mileageCart: '/mileage/shop/cart',
  assessments: '/assessments',
  myPage: '/my-page',
  settings: '/settings',
  forms: '/forms',
  qualExams: '/qual-exams',
  seating: '/seating',

  // 관리자 셸
  admin: '/admin',
  adminCohorts: '/admin/cohorts',
  adminCohortsCreate: '/admin/cohorts/create',
  adminStudents: '/admin/students',
  adminStudentsCreate: '/admin/students/create',
  adminInstructors: '/admin/instructors',
  adminInstructorsCreate: '/admin/instructors/create',
  adminAttendance: '/admin/attendance',
  adminSeatPresence: '/admin/seat-presence',
  adminSeating: '/admin/seating',
  adminAssessments: '/admin/assessments',
  adminRecords: '/admin/records',
  adminResumes: '/admin/resumes',
  adminFormTasks: '/admin/form-tasks',
  adminFormTasksCreate: '/admin/form-tasks/create',
  adminStudyRoom: '/admin/study-room',
  adminStudyRoomCreate: '/admin/study-room/create',
  adminBoard: '/admin/board',
  adminBoardNoticeCreate: '/admin/board/create',
  adminBoardScheduledCreate: '/admin/board/scheduled/create',
  adminBoardAlertPopupCreate: '/admin/board/alert-popups/create',
  adminMileage: '/admin/mileage',
  adminMileageProducts: '/admin/mileage/products',
  adminMileageProductsCreate: '/admin/mileage/products/create',
  adminMileageRequests: '/admin/mileage/requests',
  adminMileageAdjust: '/admin/mileage/adjust',
  adminMileageSettings: '/admin/mileage/settings',
  adminAiQuality: '/admin/ai-quality',
  adminMyPage: '/admin/my-page',
  adminSettings: '/admin/settings',

  // 강사 셸
  instructor: '/instructor',
  instructorResumes: '/instructor/resumes',
  instructorBoard: '/instructor/board',
  instructorBoardCreate: '/instructor/board/create',
  instructorAssessments: '/instructor/assessments',
  instructorAssessmentsCreate: '/instructor/assessments/create',
  instructorCurriculum: '/instructor/curriculum',
  instructorMyPage: '/instructor/my-page',
  instructorSettings: '/instructor/settings',
} as const;

// 파라미터가 붙는 경로들 — Flutter의 같은 이름 함수들
export const resumeEditPath = (
  resumeId: string,
  opts: { section?: string; feedback?: boolean } = {},
): string => {
  const params = new URLSearchParams();
  if (opts.section !== undefined && opts.section !== '') params.set('section', opts.section);
  if (opts.feedback === true) params.set('feedback', '1');
  const query = params.toString();
  return `/resume/${resumeId}/edit${query === '' ? '' : `?${query}`}`;
};

export const assessmentTakePath = (id: string) => `/assessments/${id}/take`;
export const assessmentResultPath = (id: string) => `/assessments/${id}/result`;

export const studyRoomNoteSourcePath = (sourceId: string) => `/study-room/notes/${sourceId}`;

export const instructorBoardNoticeEditPath = (id: string) => `/instructor/board/${id}/edit`;
export const instructorAssessmentDetailPath = (id: string) => `/instructor/assessments/${id}`;
export const instructorAssessmentEditPath = (id: string) => `/instructor/assessments/${id}/edit`;
export const instructorAssessmentSubmissionPath = (id: string, submissionId: string) =>
  `/instructor/assessments/${id}/submissions/${submissionId}`;

export const adminAssessmentDetailPath = (id: string) => `/admin/assessments/${id}`;
export const adminStudentDetailPath = (uid: string) => `/admin/students/${uid}`;
export const adminStudentEditPath = (uid: string) => `/admin/students/${uid}/edit`;
export const adminCohortEditPath = (id: string) => `/admin/cohorts/${id}/edit`;
export const adminFormTaskEditPath = (id: string) => `/admin/form-tasks/${id}/edit`;
export const adminBoardNoticeEditPath = (id: string) => `/admin/board/${id}/edit`;
export const adminBoardScheduledEditPath = (id: string) => `/admin/board/scheduled/${id}/edit`;
export const adminBoardAlertPopupEditPath = (id: string) => `/admin/board/alert-popups/${id}/edit`;
export const adminStudyRoomPackagePath = (id: string) => `/admin/study-room/${id}`;
export const adminMileageProductEditPath = (id: string) => `/admin/mileage/products/${id}/edit`;

export function homeFor(role: UserRole): string {
  switch (role) {
    case 'admin':
      return RoutePaths.admin;
    case 'instructor':
      return RoutePaths.instructor;
    default:
      return RoutePaths.dashboard;
  }
}
